.github/workflows/entropy-check.yml

```diff
@@ -22,24 +22,24 @@ jobs:
       - name: Install python env
         uses: ./.github/actions/install-python-env
         with:
-          fetch-depth: 0 # Ensure the action respects the desired fetch depth
+          fetch-depth: 0
 
         # Determine the PR branches and set environment variables
-      - name: Set PR Branch Environment Variables
-        env:
-          PR_BRANCH: ${{ github.head_ref }}
-          MAIN_BRANCH: ${{ github.base_ref }}
+      - name: Set and Display PR Branch Environment Variables
         run: |
-          "echo 'PR_BRANCH=${PR_BRANCH}' >> $GITHUB_ENV"
-          "echo 'MAIN_BRANCH=${MAIN_BRANCH}' >> $GITHUB_ENV"
+          export PR_BRANCH=${{ github.head_ref }}
+          export MAIN_BRANCH=${{ github.base_ref }}
+          echo "PR_BRANCH is $PR_BRANCH"
+          echo "MAIN_BRANCH is $MAIN_BRANCH"
```

Comment:
When running these commands together like this you typically need to use `&&` or `||` to help them chain. [See here](https://unix.stackexchange.com/a/24685) for more information on how these work. I think the current version of this attempts to run something like: `export PR_BRANCH=${{ github.head_ref }} export MAIN_BRANCH=${{ github.base_ref }} echo "PR_BRANCH is $PR_BRANCH" echo "MAIN_BRANCH is $MAIN_BRANCH"` which I believe will result in errors (the first export command will interpret everything else as flags or arguments). The newlines might also get added in here due to the `|` string continuation within yaml. I think the GitHub Actions job is maybe truncating the output, which makes it a bit confusing to look at.

Consider the following:

You can always view the exit status code of previous commands with: `echo $?`. Each command generally should exit with a zero, which means no errors or warnings occurred (be careful here, sometimes the process might "want" to find errors successfully and exit with a 0 because it did it's job successfully). If it doesn't exit with zero we generally intercept that as an error. We can also simplify this with `True` and `False` in bash to simulate what happens to commands.

```bash
# 1 "and" 0
False && True
echo $?
# shows 1

# 0 "and" 0
True && True
echo $?
# shows 0

# 1 "or" 0
False || True
echo $?
# shows 0

# 0 "or" 0
True || True
echo $?
# shows 0
```

GitHub Actions I think will check the status of the steps and because it sees the `export` didn't finish successfully (with a 0) it gathers this up for us in the job and tells us something went wrong.

Reply:
Looking at this again, I think I'm wrong here about the need for the `&&`! Leaving this up as there's still some useful information here just in case. I'm going to ponder this a bit more, apologies.